### PR TITLE
Adds a task source and uses it for resolving promises.

### LINKED ISSUE #229: Missing tasks in parallel steps in Media Capabilities
While crawling [Media Capabilities](https://w3c.github.io/media-capabilities/), the following algorithms fire an event, or resolve or reject a Promise, within a step that runs [in parallel](https://html.spec.whatwg.org/multipage/infrastructure.html#in-parallel) without first queuing a task:
* [ ] The algorithm that starts with "The decodingInfo() method method MUST run the following steps:" resolves/rejects a promise directly in a step that runs in parallel
* [ ] The algorithm that starts with "The encodingInfo() method MUST run the following steps:" resolves/rejects a promise directly in a step that runs in parallel

See [Dealing with the event loop](https://html.spec.whatwg.org/multipage/webappapis.html#event-loop-for-spec-authors) in the HTML specification for guidance on how to deal with algorithm sections that run *in parallel*.

<sub>Cc @dontcallmedom @tidoust</sub>

<sub>This issue was detected and reported semi-automatically by [Strudy](https://github.com/w3c/strudy/) based on data collected in [webref](https://github.com/w3c/webref/).</sub>

### PR DESCRIPTION
**Pull request by markafoltz**
Closes #229 by adding a media capabilities task source, and using it to resolve the promises created 
by `decodingInfo()` and `encodingInfo()`.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/media-capabilities/pull/232.html" title="Last updated on Oct 4, 2024, 11:55 PM UTC (8cd3baf)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/media-capabilities/232/6b47236...8cd3baf.html" title="Last updated on Oct 4, 2024, 11:55 PM UTC (8cd3baf)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -1077,15 +1101,23 @@ spec: mimesniff; urlPrefix: https://mimesniff.spec.whatwg.org/#
...
         </li>
         <li>
-          <a>In parallel</a>, run the <a>Create a
-          MediaCapabilitiesDecodingInfo</a> algorithm with
-          <var>configuration</var> and resolve <var>p</var> with its result.
+          Return <var>p</var>.
         </li>
         <li>
-          Return <var>p</var>.
+          <a>In parallel</a>, run following steps:
```
Comment (tidoust) [inline index.bs:1110]: Missing "the", and for consistency with the other occurrence, I would suggest

```suggestion
          Run the following steps <a>in parallel</a>:
```

File: index.bs [hunk 2]
```diff
@@ -1105,20 +1140,28 @@ spec: mimesniff; urlPrefix: https://mimesniff.spec.whatwg.org/#
...
+               its result.
+             </li>
+           </ol>
         </li>
       </ol>
     </p>
-
-  </section>
+    </section>
+  </Section>
```
Comment (tidoust) [inline index.bs:1164]:
```suggestion
  </section>
```

File: index.bs [hunk 3]
```diff
@@ -1077,15 +1101,23 @@ spec: mimesniff; urlPrefix: https://mimesniff.spec.whatwg.org/#
...
         </li>
         <li>
-          Let <var>p</var> be a new promise.
+          Let <var>p</var> be a new Promise.
         </li>
         <li>
-          <a>In parallel</a>, run the <a>Create a
-          MediaCapabilitiesDecodingInfo</a> algorithm with
-          <var>configuration</var> and resolve <var>p</var> with its result.
+          Return <var>p</var>.
```
Comment (tidoust) [inline index.bs:1107]: If I read [Infra](https://infra.spec.whatwg.org/#algorithm-control-flow) correctly, a "return" statement terminates the algorithm. This step and the following one should rather be swapped.

File: index.bs [hunk 4]
```diff
@@ -1105,20 +1140,28 @@ spec: mimesniff; urlPrefix: https://mimesniff.spec.whatwg.org/#
...
         </li>
         <li>
-          Let <var>p</var> be a new promise.
+          Let <var>p</var> be a new Promise.
         </li>
         <li>
-          <a>In parallel</a>, run the <a>Create a MediaCapabilitiesEncodingInfo</a>
-          algorithm with <var>configuration</var> and resolve <var>p</var>
-          with its result.
+          Return <var>p</var>.
```
Comment (tidoust) [inline index.bs:1146]: Same comment as above re. ordering of steps.

File: index.bs [hunk 5]
```diff
@@ -1105,20 +1140,28 @@ spec: mimesniff; urlPrefix: https://mimesniff.spec.whatwg.org/#
...
         </li>
         <li>
-          Return <var>p</var>.
+          Run the following steps <a>in parallel</a>.
```
Comment (tidoust) [inline index.bs:1149]:
```suggestion
          Run the following steps <a>in parallel</a>:
```

File: index.bs [hunk 6]
```diff
@@ -39,6 +43,10 @@ spec: mediastream-recording; urlPrefix: https://www.w3.org/TR/mediastream-record
 
 spec: mimesniff; urlPrefix: https://mimesniff.spec.whatwg.org/#
     type: dfn; text: valid mime type; url: valid-mime-type
+
+spec: ecmascript; urlPrefix: https://tc39.es/ecma262/multipage/
+    type: dfn;
+        text: current realm; url: executable-code-and-execution-contexts.html#sec-code-realms
```
Comment (tidoust) [inline index.bs:49]: I don't have anything better to suggest right away but I don't like it that we have to create an anchor for something as common as queueing a task. I'll try to think at better ways to formulate things. In the meantime, that looks good enough ;)

**Review by tidoust — CHANGES_REQUESTED** [review]

File: index.bs (hunk 6 above)
Comment (tidoust) [reply index.bs:49]: Actually, I have something to suggest: reference the actual term defined in ECMAScript at the given fragment: "the current realm record", which can be referenced with `[=the current realm record=]` without having to introduce a new anchor.

File: index.bs [hunk 7]
```diff
@@ -1052,8 +1060,24 @@ spec: mimesniff; urlPrefix: https://mimesniff.spec.whatwg.org/#
...
+      The [=task source=] for the tasks mentioned in this specification
+      is the <dfn>media capabilities task source</dfn>.
+    </p>
+
     <p>
-      The {{decodingInfo()}} method method MUST run the following steps:
+      When an algorithm <dfn lt="queue a Media Capabilities task">queues
+      a Media Capabilities task</dfn> <var>T</var>, the user agent
+      MUST [=queue a global task=] <var>T</var> on the [=media capabilities
+      task source=] using the [=global object=] of the [=current realm=].
```
Comment (tidoust) [inline index.bs:1074]:
```suggestion
      task source=] using the [=global object=] of [=the current realm record=].
```

File: index.bs (hunk 6 above)
Comment (markafoltz) [reply index.bs]: Done

File: index.bs (hunk 7 above)
Comment (markafoltz) [reply index.bs]: Done

File: index.bs (hunk 3 above)
Comment (markafoltz) [reply index.bs]: Done

File: index.bs (hunk 1 above)
Comment (markafoltz) [reply index.bs]: Done

File: index.bs (hunk 4 above)
Comment (markafoltz) [reply index.bs]: Done

File: index.bs (hunk 5 above)
Comment (markafoltz) [reply index.bs]: Done

File: index.bs (hunk 2 above)
Comment (markafoltz) [reply index.bs]: Done

**Review by tidoust — APPROVED** [review]

**Review by chrisn — APPROVED** [review]

**Pull request merged**